src/unix_process/unix_process_runtime.c

```diff
@@ -85,25 +85,33 @@ void halt(char *format, ...)
     exit(2);
 }
 
-#ifndef __aarch64__
-static heap allocate_tagged_region(heap h, u64 tag)
-{
-    u64 size = 256 * MB;
-    void *region = mmap(pointer_from_u64(tag << VA_TAG_OFFSET),
-                        size, PROT_READ | PROT_WRITE, MAP_FIXED | MAP_ANON | MAP_PRIVATE, -1, 0);
-    return (heap)create_id_heap(h, h, u64_from_pointer(region), size, 1, false);
-}
-#else
 struct tagheap {
     struct heap h;
     heap mh;
-    u64 vtag;
+    u64 tag;
 };
 
+#ifndef __aarch64__
+
+static inline u64 tag_alloc(heap h, bytes length)
+{
+    struct tagheap *th = (struct tagheap *)h;
+    u64 base = allocate_u64(th->mh, length + 1);
+    return base + 1;
```

Comment:
The tag is not being inserted here; need *(u8*)pointer_from_u64(base) = th->tag.

I found this same bug in src/boot/boot.h:tagged_allocate.

Reply:
Right, this is now fixed in https://github.com/nanovms/nanos/pull/1452/commits/27f12b82d5401d9acdd53bc44ae753bcdb376e14. (I also changed the extra space for the tag to be 8 bytes there too, so that bootloaders running in long mode don't have to do unaligned memory accesses.)